pkg/component/monitoring/prometheus/component.go

```diff
@@ -88,6 +92,20 @@ type CentralConfigs struct {
 	PodMonitors []*monitoringv1.PodMonitor
 }
 
+// IngressValues contains configuration for exposing this Prometheus instance via an Ingress resource.
+type IngressValues struct {
+	// AuthSecretName is the name of the auth secret.
+	AuthSecretName string
+	// Host is the hostname under which the Prometheus instance should be exposed.
+	Host string
+	// SecretsManager is the secrets manager used for generating the TLS certificate if no wildcard certificate is
+	// provided.
+	SecretsManager secretsmanager.Interface
```

Comment:
I understand your surprise 😄 I was struggling a bit here as well.
In the end, the motivation was that `Ingress` is optional. Hence, the secrets manager will not be needed in all cases. When it was not a field in the values, it would be necessary to always provide it, even for the cases where no `Ingress` is deployed.
If you prefer to move it, I can do so.

Reply:
I guess I understand your reasoning. It is a little different compared to most other components, but most components also do not have so many slightly different use cases.
The current approach works for me. `prometheus` stays a bit of a unicorn, but that fits well.